File: Arrays & Strings/002 Strings/Ransom Note/Solution.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int count[]=new int[26];
                
        for(int i=0;i<magazine.length();i++)
            count[magazine.charAt(i)-'a']++;
        
        for(int i=0;i<ransomNote.length();i++)
            //If ransomNote contains any character or more number of character than magazine
            //it cannot be constructed
           if(--count[ransomNote.charAt(i)-'a']<0)
               return false;

        return true;
    }
}

File: Arrays & Strings/002 Strings/Ransom Note/Solution.kt
class Solution {
    fun canConstruct(ransomNote: String, magazine: String): Boolean {
        val count= IntArray(26)
                
        for(i in magazine)
            count[i-'a']++
        
        for(i in ransomNote)
            //If ransomNote contains any character or more number of character than magazine
            //it cannot be constructed
           if(--count[i-'a']<0)
               return false

        return true
    }
}
